Assets/MixedRealityToolkit.Services/InputAnimation/Editor/InputRecordingWindow.cs

```diff
@@ -0,0 +1,351 @@
+        }
+
+        private string loadedFilePath = "";
+
+        private IMixedRealityInputRecordingService recordingService = null;
+        private IMixedRealityInputRecordingService RecordingService =>
+            recordingService ??
+            (recordingService = MixedRealityToolkit.Instance.GetService<IMixedRealityInputRecordingService>());
+
+        private IMixedRealityInputPlaybackService playbackService = null;
+        private IMixedRealityInputPlaybackService PlaybackService =>
+            playbackService ??
+            (playbackService = MixedRealityToolkit.Instance.GetService<IMixedRealityInputPlaybackService>());
+
+        private IInputSimulationService simService = null;
+        private IInputSimulationService SimService =>
+            simService ??
+            (simService = MixedRealityToolkit.Instance.GetService<IInputSimulationService>());
+
+        public enum ToolMode
+        {
+            /// <summary>
+            /// Record input animation and store in the asset.
+            /// </summary>
+            Record,
+            /// <summary>
+            /// Play back input animation as simulated input.
+            /// </summary>
+            Playback,
+        }
+
+        private ToolMode mode = ToolMode.Record;
+        public ToolMode Mode
+        {
+            get { return mode; }
+            private set
+            {
+                mode = value;
+            }
+        }
+
+        /// Icon textures
+        private Texture2D iconPlay = null;
+        private Texture2D iconPause = null;
+        private Texture2D iconRecord = null;
+        private Texture2D iconRecordActive = null;
+        private Texture2D iconStop = null;
+        private Texture2D iconStepFwd = null;
+        private Texture2D iconStepBack = null;
+        private Texture2D iconJumpFwd = null;
+        private Texture2D iconJumpBack = null;
+
+        [MenuItem("Mixed Reality Toolkit/Utilities/Input Recording")]
+        private static void ShowWindow()
+        {
+            InputRecordingWindow window = GetWindow<InputRecordingWindow>();
+            window.titleContent = new GUIContent("Input Recording");
+            window.minSize = new Vector2(380.0f, 680.0f);
+            window.Show();
+        }
+
+        private void OnGUI()
+        {
+            LoadIcons();
+
+            string[] modeStrings = Enum.GetNames(typeof(ToolMode));
+            Mode = (ToolMode)GUILayout.SelectionGrid((int)Mode, modeStrings, modeStrings.Length);
+
+            switch (mode)
+            {
+                case ToolMode.Record:
+                    DrawRecordingGUI();
+                    break;
+                case ToolMode.Playback:
+                    DrawPlaybackGUI();
+                    break;
+            }
+
+            EditorGUILayout.Space();
+
+// XXX Reloading the scene is currently not supported,
+// due to the life cycle of the MRTK "instance" object.
+// Enable the button below once scene reloading is supported!
```

Comment:
Could we please provide a bug reference here?

Reply:
I have added a ref to #4530 now